src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ActionPlanController.kt

```diff
@@ -101,4 +104,15 @@ class ActionPlanController(
     val actionPlan = actionPlanService.getActionPlanByReferral(referralId)
     return ActionPlanDTO.from(actionPlan)
   }
+
+  @PostMapping("/action-plan/{id}/approve")
+  fun approveActionPlan(@PathVariable id: UUID, authentication: JwtAuthenticationToken): ActionPlanDTO {
+    val user = userMapper.fromToken(authentication)
+
+    if (!userTypeChecker.isProbationPractitionerUser(user)) {
+      throw AccessError(user, "could not approve action plan", listOf("only probation practitioners can approve action plans"))
```

Comment:
we don't have these kinda checks everywhere, should we?

Reply:
I think it is necessary otherwise an SP could do it if they knew the link?

Reply:
Can it be approved by any PP?

Reply:
we'll have to follow up on that... for now let's say yes.

Reply:
nit: We are using userTypeChecker in ReferralService but not ReferralController. I think it would be best to be consistent (I think controller is more suitable)

Reply:
sure thing - i'll move it